documentation/src/main/asciidoc/reference/introduction.adoc

```diff
@@ -1053,10 +1053,25 @@ Hibernate Reactive session associated with the context.
 === Vert.x instance service
 
 The `VertxInstance` service defines how Hibernate Reactive obtains an instance
-of Vert.x. The default implementation just creates one the first time it's
-needed. But if your program requires control over how the Vert.x instance is
-created, or how it's obtained, you can override the default implementation and
-provide your own `VertxInstance`. Let's consider this example:
+of Vert.x when there is no instance associated with the calling thread. The
+default implementation just creates one the first time it's needed. But if your
+program requires control over how the Vert.x instance is created, or how it's
+obtained, you can override the default implementation and provide your own
+`VertxInstance`.
+
```

Comment:
I'm not sure about anything written after this point.

I think there are only two things to mention:
* Hibernate Reactive will bind the to the current Vertx instance if the factory is created inside an existing context (and will log a message about it)
* Hibernate Reactive will create a new instance if none is detected when the factory is created

The user needs to do something only if this is not the behaviour they want.
There are many way to create a factory inside a context without using `runOnContext`.

Reply:
Ah sorry, I mean it logs the message when a new Vertx instance is created is created

Reply:
We should also mention that when Hibernate Reactive creates a new instance of Vert.x, it will also shut it down when the factory gets closed.

Reply:
See what I documented is how I actually want it to work.

But you're right: it doesn't work like that today.

Reply:
> how I actually want it to work.

Well, naively.